File: rust/examples/test_basic.rs
use emtrace::{expect, init, trace, traceln};

fn main() {
    expect!(
        "Hello from the basic test!\n\
         An integer: 42\n\
         A string: a string\n"
    );
    init(&mut std::io::stdout().lock()).unwrap();

    traceln!("Hello from the basic test!");
    let x: i32 = 42;
    traceln!("An integer: {}", i32: x);
    let s: &str = "a string";
    trace!("A string: ");
    traceln!("{}", str: s);
    // traceln!("A list: {}", [i32]: vec![1, 2, 3, 4]);
}

File: c/examples/test_basic.c
#include "test_utils.h"
#include <emtrace/emtrace.h>

EXPECT_OUTPUT(
    "Hello from the basic test!\n"
    "An integer: 42\n"
    "A string: a string\n"
);

#ifdef TEST_COBS
EMTRACE_MAGIC_COBS(size_t)
#else
EMTRACE_MAGIC(size_t)
#endif

int main(void) {
    emtrace_init();
    EMTRACELN("Hello from the basic test!");
    int x = 42;
    EMTRACELN_F("An integer: {}", VAL(int, x));
    const char* s = "a string";
    EMTRACELN_F("A string: {}", STR(s));
    return 0;
}
